addons/website_slides/static/src/js/slides_course_quiz.js

```diff
@@ -172,6 +173,60 @@ odoo.define('website_slides.quiz', function (require) {
                 this.$el.append(QWeb.render('slide.slide.quiz.finish', {'widget': this}));
                 $modal = this.$('#slides_quiz_modal');
             }
+            var self = this;
+            $modal.on('shown.bs.modal', function () {
+                self.$('[data-toggle="tooltip"]').tooltip({
+                    trigger: 'manual',
+                    container: '.progress-bar-tooltip',
+                    boundary: ''
+                }).tooltip('show');
+                var rank = self.quiz.rankProgress;
+                // Animate a counter from old karma value to new karma value
+                $modal.find('.tooltip-inner').prop('Counter', rank.oldKarma).animate({
+                    Counter: rank.newKarma
+                }, {
+                    duration: 1500,
+                    step: function (now) {
+                        $modal.find('.tooltip-inner').text(Math.ceil(now));
+                        self.$('[data-toggle="tooltip"]').tooltip('update');
+                    }
+                });
+                if (rank.levelUp) {
+                    $modal.find('.o_wslides_quiz_modal_title').text('You Level Up!');
+                    $modal.find('.o_wslides_quiz_modal_title').css('font-size', '3rem');
+                    $modal.find('.progress-bar').animate({
+                        width: '100%'
+                    }, 750, function () {
+                        $modal.find('.o_wslides_rank_motivational').animate({
+                            opacity: 0
+                        }, 600, function () {
+                            $modal.find('.o_wslides_rank_motivational').html(rank.description);
+                            $modal.find('.o_wslides_rank_motivational').animate({
+                                opacity: 1
+                            }, 600);
+                        });
+                        $modal.find('.progress-bar').animate({
+                            width: '0%'
+                        }, 0, function () {
+                            $modal.find('.o_wslides_rank_lower_bound').html(rank.newLowerBound);
+                            $modal.find('.o_wslides_rank_upper_bound').html(rank.lastRank ? rank.newKarma : rank.newUpperBound);
+                            $modal.find('.progress-bar').animate({
+                                width: rank.newProgress + '%'
+                            }, 750);
+                        });
+                    });
+                } else {
+                    $modal.find('.progress-bar').animate({
+                        width: rank.newProgress + '%'
+                    }, 1500);
+                }
+                $modal.find('h4.o_wslides_xp_gained').delay(1000).animate({
+                    opacity: 1
+                }, 500);
+                setTimeout(function () {
+                   $modal.find('.o_wslides_quiz_dismiss_btn').removeClass('d-none');
+                }, 1500);
```

Comment:
This whole block feels like a lot of magic...

There are functions and over-nesting and random numbers everywhere.
It would probably help if you split into private function (eg: `self_animateProgressBar($modal.find('.progress-bar'))`)
You could add doc on each private sub-method to explain what you're doing exactly.

Also maybe naming your timers would make it easier to understand (eg: `var xpGainDelay = 500`) but not sure about that one, I need to see how it looks.

Doing that will ease code reading and maintenance (because if I tell you to dive back into this code in 6 months... well good luck :D )

Reply:
I made two private functions to distinguish the progress bar animation and the others animations
But I'm not sure where to declare the delay variables to use them in both functions...